### PR TITLE
Send auto opt out response

### PR DESCRIPTION
**Pull request by agreenspan24**
# Fixes

https://github.com/MoveOnOrg/Spoke/issues/2009

#### Problem
The auto opt-out message handler saves an opt out to our database. However, it doesn't let the contact know that we've removed them from our list. Plus, texters don't see the auto opted-out contact, so they can't send an opt-out message manually.

#### Solution
As part of the auto opt-out message handler, send the organization's default opt out message back to the contact being opted out.

#### Context
Twilio has their own effective auto opt-out, which blocks any additional messages after specific keywords are sent. We may need to exempt these specific keywords from this new feature.
https://support.twilio.com/hc/en-us/articles/223134027-Twilio-support-for-opt-out-keywords-SMS-STOP-filtering-

# Checklist:

- [x] I have manually tested my changes on desktop and mobile
- [x] The test suite passes locally with my changes
- [x] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [x] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer
- [x] I have made any necessary changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] My PR is labeled [WIP] if it is in progress

### CODE REVIEW

File: docs/REFERENCE-environment_variables.md [hunk 1]
```diff
@@ -97,6 +97,7 @@
...
 | ROLLBAR_ENDPOINT                                 | Endpoint URL for Rollbar error tracking.                                                                                                                                                                                                                                                                                                                                                    |
+| SEND_AUTO_OPT_OUT_RESPONSE                           | Send the organization's default opt-out message if a user is auto-opted out. Alternatively, set a shouldAutoRespond property in an item for the auto opt-out regex to true, for it to only apply on those regex matches.                                                                |
```
Comment (crayolakat) [inline docs/REFERENCE-environment_variables.md:100]: `Alternatively, set a shouldAutoRespond property in an item for the auto opt-out regex to true, for it to only apply on those regex matches.`

Won't the setting the SEND_AUTO_OPT_OUT_RESPONSE also only apply on the regex matches in the auto opt-out regex?

**Review by crayolakat — COMMENTED** [review]
Looks great! I had 1 small question which I left in the comments.

File: docs/REFERENCE-environment_variables.md (hunk 1 above)
Comment (agreenspan24) [reply docs/REFERENCE-environment_variables.md:100]: So if I remember correctly, we envisioned two paths: 
* For those using the default auto opt out regex, or those who wanted to send the message for every auto opt out, they could just set SEND_AUTO_OPT_OUT_RESPONSE to true and be done. 
* For those who only wanted to send the message for some of their opt outs (for example, not respond to the hostile opt outs), they could customize the behavior per auto opt out regex with the shouldAutoRespond property and set SEND_AUTO_OPT_OUT_RESPONSE to false. 

Does that answer your question?

Comment (crayolakat) [reply docs/REFERENCE-environment_variables.md:100]: Yes, I just discovered that you can have multiple regex for the customized auto-optouts :)

**Review by crayolakat — APPROVED** [review]

**Comment by crayolakat** [comment]
This was merged into the `main` branch for the 12.2 release, but some rebasing caused the PR to not close automatically. Manually closing the PR.

**Pull request closed without merging**